### ISSUE TITLE
Event for reconnection giving up

### ISSUE THREAD

**Issue by PaquitoSoft** [NodeRedis/node_redis#586]
Hello:

I think it would be great if the client would emit an event when it gives up trying to reconnect to a dead Redis instance because a user configured it to do so using _connect_timeout_ or _max_attempts_ attributes.

I mean updating this code:
```
if (self.connect_timeout && self.retry_totaltime >= self.connect_timeout) {
    self.retry_timer = null;
    // TODO - engage Redis is Broken mode for future commands, or whatever
    console.error("node_redis: Couldn't get Redis connection after " + self.retry_totaltime + "ms.");
    return;
}
```

to look something like this:
```
if (self.connect_timeout && self.retry_totaltime >= self.connect_timeout) {
    self.retry_timer = null;
    self.emit('reconnectionGiveup');
    return;
}
```

and the same for the _max_attemps_ case.

**Comment by aryehischechter** [comment]
+1
Will this not happen because of the reconnect restructuring?

**Comment by blainsmith** [comment]
In order to better support this project and its new group of collaborators under a new org we are trying to clean up and close issues older than 6 months. Your issue may have been fixed since the issue was open, however, if you are still experiencing a problem **please** re-open this issue and we will label it accordingly.

**Comment by BridgeAR** [comment]
This is going to be fixed by #829
